mage_ai/streaming/sources/rabbitmq.py

```diff
@@ -0,0 +1,98 @@
+from mage_ai.shared.config import BaseConfig
+from mage_ai.streaming.sources.base import BaseSource
+from dataclasses import dataclass
+from typing import Callable, Dict
+from collections import namedtuple
+import pika
+from pika.exceptions import AMQPConnectionError
+
+@dataclass
+class ConsumeConfig:
+    auto_ack: bool = False
+    exclusive: bool = False
+    inactivity_timeout: float = None
+
+
+
+
+
+@dataclass
+class PikaConfig(BaseConfig):
+    connection_host: str
+    connection_port: int
+    queue_name: str
+    configure_consume: bool = False
+    username: str = 'guest'
+    password: str = 'guest'
+    amqp_url_virtual_host: str = r'%2f'
+    consume_config = ConsumeConfig = None
+
+    @classmethod
+    def parse_config(self, config: Dict) -> Dict:
+        consume_config = config.get('consume_config')
+        if consume_config is not None and type(consume_config) is dict:
+            config['consume_config'] = ConsumeConfig(**consume_config)
+        return config
+
+    
+
+
+class PikaSource(BaseSource):
```

Comment:
I would just call it RabbitMQSource and RabbitMQConfig.

Reply:
Fixed on new commit (57b943b2dfdcf4d1ab128bf663facbd64e20beee)